autoload/neomake.vim

```diff
@@ -661,6 +708,31 @@ function! s:RegisterJobOutput(jobinfo, lines) abort
     endif
 endfunction
 
+function! s:GetChannelId(channel) abort
+    return ch_info(a:channel)['id']
+endfunction
+
+function! neomake#MakeHandlerVimStdout(channel, output) abort
+    call neomake#utils#DebugMessage('MakeHandlerVim: stdout: ' . a:channel)
+    let channel_id = s:GetChannelId(a:channel)
+    let jobinfo = s:jobs[channel_id]
+    call neomake#MakeHandler(channel_id, split(a:output, "\n", 1), 'stdout')
+endfunction
+
+function! neomake#MakeHandlerVimStderr(channel, output) abort
+    call neomake#utils#DebugMessage('MakeHandlerVim: stderr: ' . a:channel)
+    let channel_id = s:GetChannelId(a:channel)
+    let jobinfo = s:jobs[channel_id]
+    call neomake#MakeHandler(channel_id, split(a:output, "\n", 1), 'stderr')
+endfunction
+
+function! neomake#MakeHandlerVimClose(channel) abort
+    call neomake#utils#DebugMessage('Channel has been closed: ' . a:channel)
+    let channel_id = s:GetChannelId(a:channel)
+    let jobinfo = s:jobs[channel_id]
+    call neomake#MakeHandler(channel_id, job_info(jobinfo.job)['exitval'], 'exit')
+endfunction
```

Comment:
This part really smells like it should be refactored.. even when it's just having one function and using a partial (see `:h Partial`) for stdout/stderr.
Also I am not sure if all the channel/job mapping is needed.
Messing with this should be easy now since tests are good again.. :)